components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIStoreImpl.java

```diff
@@ -662,23 +654,34 @@ public void completeWorkflow(WorkflowExecutor workflowExecutor, Workflow workflo
         } else {
             String message = "Invalid workflow type:  " + workflow.getWorkflowType();
             log.error(message);
-            throw new WorkflowException(message);
+            throw new APIManagementException(message, ExceptionCodes.WORKFLOW_EXCEPTION);
         }
     }
     
-    private void updateWorkflowEntries(Workflow workflow) throws APIMgtDAOException {
+    private void updateWorkflowEntries(Workflow workflow) throws APIManagementException {
         workflow.setUpdatedTime(LocalDateTime.now());
-        getWorkflowDAO().updateWorkflowStatus(workflow);
-        //TODO stats stuff
+        try {
+            getWorkflowDAO().updateWorkflowStatus(workflow);
+            //TODO stats stuff
+        } catch (APIMgtDAOException e) {
+            String message = "Error while updating workflow entry";
```

Comment:
workflow object will have 'workflowReference' which won't be much of a use since it is an internal id. But the exception will have the error message related to the exception so i think that would be enough to find the error